include/threads.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Mutexes/Condvars
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _PHOENIX_THREADS_H_
+#define _PHOENIX_THREADS_H_
+
+
+/* Mutex attributes */
+
+/* clang-format off */
+enum { LOCK_TYPE_NORMAL = 0, LOCK_TYPE_DEFAULT = LOCK_TYPE_NORMAL, LOCK_TYPE_RECURSIVE, LOCK_TYPE_ERRORCHECK };
```

Comment:
I have two problems with these:
1. MISRA Rule 8.12, enums should be unique
2. upper case is reserved for macros

My proposition is to use defines for these